### PR TITLE
[DOCS] Adapted docs for new intelmq-api

### PR DESCRIPTION
**Pull request by waldbauer-certat**
Changes for IntelMQ Docs in order to work with the newer version including the intelmq-api.

### CODE REVIEW

File: docs/dev/release-procedure.rst [hunk 1]
```diff
@@ -64,6 +64,18 @@ First, test all the steps first with the `unstable-repository <http://build.open
...
 
+************
+Docker Image
+************
+
+Releasing a new Docker image is very easy.
+
+* Clone `IntelMQ Docker Repository <https://github.com/certat/intelmq-docker>`_
+* Set `chmod +x build.sh`, `chmod +x test.sh` & `chmod +x publish.sh`
+* Run `sudo ./build.sh`, check your console if the build was successful.
```
Comment (schacht-certat) [inline docs/dev/release-procedure.rst:75]:
```
vagrant@buster:~/intelmq-docker$ sudo ./build.sh 
./build.sh: line 3: cd: ../intelmq: No such file or directory
./build.sh: line 4: cd: ../intelmq-manager: No such file or directory
./build.sh: line 5: cd: ../intelmq-api: No such file or directory
Building new IntelMQ-Image v1.0
Core :
Manager :
Api :
Build_date: 2021-02-04T09:02:57Z
python3: can't open file 'setup.py': [Errno 2] No such file or directory
unable to prepare context: unable to evaluate symlinks in Dockerfile path: lstat /home/vagrant/intelmq-docker/intelmq-manager/.docker: no such file or directory
unable to prepare context: unable to evaluate symlinks in Dockerfile path: lstat /home/vagrant/intelmq-docker/intelmq-manager/.docker: no such file or directory
```

Comment (schacht-certat) [reply docs/dev/release-procedure.rst:75]: Are you sure that the build step has to be run with root privileges?

File: docs/dev/release-procedure.rst [hunk 2]
```diff
@@ -64,6 +64,18 @@ First, test all the steps first with the `unstable-repository <http://build.open
...
+* Run `sudo ./test.sh` - It will run nosetests3 with exotic flag. All errors/warnings will be displayed.
+* If no error/warning was shown, you can release with `sudo ./publish.sh` **ATTENTION** Only available for CERT.AT employee's.
```
Comment (schacht-certat) [inline docs/dev/release-procedure.rst:77]: I'm pretty sure that the publishing of a container images does not require superuser privileges

**Review by schacht-certat — CHANGES_REQUESTED** [review]
Please also fix the type (see automated tests)

**Review by schacht-certat — CHANGES_REQUESTED** [review]
Please fix the typo (see automated tests) and resolve the requested changes if they are resolved

File: docs/dev/release-procedure.rst [hunk 3]
```diff
@@ -64,6 +64,17 @@ First, test all the steps first with the `unstable-repository <http://build.open
...
+************
+Docker Image
+************
+
+Releasing a new Docker image is very easy.
+
+* Clone `IntelMQ Docker Repository <https://github.com/certat/intelmq-docker>`_ with `git clone https://github.com/certat/intelmq-docker.git --recursive` as this repository contains submodules
+* Run `./build.sh`, check your console if the build was successful.
+* Run `./test.sh` - It will run nosetests3 with exotic flag. All errors/warnings will be displayed.
+* If no error/warning was shown, you can release with `./publish.sh` **ATTENTION** Only available for CERT.AT employee's.
```
Comment (schacht-certat) [inline docs/dev/release-procedure.rst:76]: I think what you want to say is that pushing to the official docker hub intelmq namespace is only allows for core contributors, but those don't have necessarily to be cert.at employees. Given that the documentation of releases makes only sense for contributors with the right permissions I think the last sentence should be removed.

File: docs/user/installation.rst [hunk 4]
```diff
@@ -168,40 +168,40 @@ Navigate to your preferred installation directory, i. e. use ``mkdir ~/intelmq &
 You need to prepare some volumes & configs. Edit the left-side after -v, to change paths.
```
Comment (schacht-certat) [inline docs/user/installation.rst:168]: Please mention that the `example_config` folder the docker instructions below refer to can be found in the `intelmq-docker` repository. Or adjust the documentation to the paragraph about docker-compose where the user is instructed to clone the repository in the first step, to get all the needed files.

File: docs/user/upgrade.rst [hunk 5]
```diff
@@ -46,19 +46,21 @@ Docker (beta)
...
 
 .. code-block:: bash
 
    docker-compose pull
 
-You can check the current versions from intelmq & intelmq-manager via git commit ref.
+You can check the current versions from intelmq & intelmq-manager & intelmq-api via git commit ref.
 
-The first one is the intelmq commit ref, and the second one is for intelmq-manager
+Version format for each included item is `key=value` and they are saparated via `,`. I. e. `IntelMQ=ab12cd34f, IntelMQ-API=xy65z23`.
```
Comment (schacht-certat) [inline docs/user/upgrade.rst:59]: **The** version format...

**Review by schacht-certat — CHANGES_REQUESTED** [review]

File: docs/user/installation.rst (hunk 4 above)
Comment (schacht-certat) [reply docs/user/installation.rst:168]: How was this resolved?

Comment (schacht-certat) [reply docs/user/installation.rst:168]: I still don't see how this was resolved. Am I missing something?

Comment (waldbauer-certat) [reply docs/user/installation.rst:168]: You mention the usage without docker-compose but the ``intelmq-docker`` repository is built for docker-compose usage. I will add that you need to clone the repository too for building & testing, but for installation its not required as you download all images directly from dockerhub. It's the not recommended way.

Comment (schacht-certat) [reply docs/user/installation.rst:168]: But in your `code-block` you give the example with `-v ~/intelmq/example_config/intelmq/etc/:/opt/intelmq/etc/ ` for **running** the docker image. The command **fails** without the `example_folder`- so either you explain to the user how to create the `example_folder` and all the needed files for **running** the docker image successfully *or* you mention that the repository needs to be cloned.

Comment (waldbauer-certat) [reply docs/user/installation.rst:168]: Easier solution is to clone the repository as you can switch to `docker-compose` instead if you want. Added the clone command now to documentation for non `docker-compose` usage too.

**Review by schacht-certat — APPROVED** [review]

**Pull request merged**